text/0000-enable-embroider.md

```diff
@@ -0,0 +1,190 @@
+
+```diff
+-return app.toTree();
++const { Webpack } = require('@embroider/webpack');
++return require('@embroider/compat').compatBuild(app, Webpack);
+```
+
+### New application flow
+
+The above changes will be applied to the blueprint for `ember new`. Additionally, the build flag option of `--embroider` will exist so that the option will be recorded in `config/ember-cli-update.json` like:
+
+```json
+{
+  "schemaVersion": "1.0.0",
+  "packages": [{
+    "name": "ember-cli",
+    "version": "X.X.X",
+    "blueprints": [{
+      "name": "app",
+      "outputRepo": "https://github.com/ember-cli/ember-new-output",
+      "codemodsSource": "ember-app-codemods-manifest@1",
+      "isBaseBlueprint": true,
+      "options": [
+        "--embroider"
+      ]
+    }]
+  }]
+}
+```
+
+This will ensure that `ember-cli-update` only applies these changes to applications that are already on Embroider.
+
+### Existing application migration flow
+
+In order to make the migration as simple as possible, a codemod will be created. The codemod CLI will provide two commands: `preflight` and `migrate`.
+
+Example usage:
+
+`npx @embroider/cli preflight`
+`npx @embroider/cli migrate`
+
+#### Preflight
+
+The goal of this command is to give users an easy yes or no answer to the question: can I start migrating to Embroider? As an example, we can check if the project is on the minimal set of package versions of known "fixed" addons along with other static checks.
+
+#### Migrate
+
+The migrate command is purely to codemod the file changes above to get your app onto Embroider.
+
+### Deprecations
+
+While Embroider strives to be as backwards compatible as possible, some existing concepts will not directly apply under Embroider. These concepts largely rely around the fact that Embroider does not want to you care about the contents of your output. This means that concepts like fingerprints asset file names or modifing output paths of your dist assets will not be supported.
+
+#### Fingerprinting
+
+In classic builds there was a `app.options.fingerprint.exclude` configuration which users could configure to disable fingerprinting for specific files or `app.options.fingerprint.enabled` field to disable fingerprinting for all files. Under Embroider, JS and CSS assets will always be fingerprinted in production mode (all app code is always fingerprinted even in dev mode as they will become "chunks") except for any asset that contains the data attribute `data-embroider-ignore`. An example of this is that `testem.js` is [excluded from fingerprinting](https://github.com/ember-cli/ember-cli/blob/1301d24c8d7dec1f1559cffd0688d46bcd50284e/lib/broccoli/ember-app.js#L351-L355) as that explicit file name is expected. However, under embroider we can change `tests/index.html` to contain:
```

Comment:
Are other fingerprint options still supported?

Reply:
Currently no. I can make this section state more explicitly that `app.options.fingerprint` are not enabled

Reply:
I think we could (should?) point out that there is still quite a gap in the functionality, as Embroider will only fingerprint the basic JS and CSS files that it adds to index.html. `broccoli-asset-rev` would fingerprint arbitrary files, e.g. images, and rewrite all the references to those (as long as it can "see" the file paths, so they have to be literals, i.e. no dynamic concatenations logic or so), e.g in JS or CSS files. AFAIK there is no solution for this yet in Embroider-land. 

Which is quite annoying as you cannot serve images with HTTP headers that enable long-term browser caching as you normally would. Or if you do, they are not invalidated when they change. 

And people are probably quite used to this (the previous behavior), so when this does not work it would feel like a bug rather than a missing feature.